datafusion-cli/src/cli_context.rs

```diff
@@ -93,6 +116,120 @@ impl CliSessionContext for SessionContext {
         &self,
         plan: LogicalPlan,
     ) -> Result<DataFrame, DataFusionError> {
-        self.execute_logical_plan(plan).await
+        SessionContext::execute_logical_plan(self, plan).await
+    }
+}
+
+/// Session context used by the CLI with memory profiling support.
+pub struct ReplSessionContext {
+    ctx: SessionContext,
+    memory_profiling: AtomicBool,
+    tracked_memory_pool: Option<Arc<dyn TrackedPool>>,
+}
+
+impl ReplSessionContext {
+    pub fn new(
+        ctx: SessionContext,
+        base_memory_pool: Arc<dyn MemoryPool>,
+        top_memory_consumers: usize,
+    ) -> Self {
+        let tracked_memory_pool = if top_memory_consumers > 0 {
```

Comment:
this seems like it is redundant with the top level pool that is installed with the CLI (which also may be a tracked pool)